server/consumer.go

```diff
@@ -116,6 +116,33 @@ type SequenceInfo struct {
 type CreateConsumerRequest struct {
 	Stream string         `json:"stream_name"`
 	Config ConsumerConfig `json:"config"`
+	Action ConsumerAction `json:"action"`
+}
+
+type ConsumerAction int
+
+const (
+	ActionCreateOrUpdate ConsumerAction = iota
+	ActionUpdate
+	ActionCreate
+)
+
+func (action *ConsumerAction) UnmarshalJSON(b []byte) error {
+	var str string
+	if err := json.Unmarshal(b, &str); err != nil {
+		return err
+	}
+	switch strings.ToLower(str) {
```

Comment:
we definitely do not want this to be foo, Foo, FOO, foO etc.  That's just impossible to validate and leads to questions.  One way is best.

Reply:
Was changed to be more graceful as a review point, but I think that you're right here.